Assets/Scripts/StandardSamples/SignInWithGoogle/SignInWithGoogleManager.cs

```diff
@@ -0,0 +1,101 @@
+        public string idToken;
+        
+        /// <summary>
+        /// Proxy class to receive Android callbacks in C#
+        /// </summary>
+        public EOSCredentialManagerCallback() : base("androidx.credentials.CredentialManagerCallback") { }
+
+        /// <summary>
+        /// Succeeding Callback of GetCredentialAsync  
+        /// GetCredentialAsync is called in com.playeveryware.googlelogin.login.SignInWithGoogle)
+        /// </summary>
+        /// <param name="credentialResponseResult"></param>
+        public void onResult(AndroidJavaObject credentialResponseResult)
+        {
+            /// Parses the response resilt into google credentials
+            loginObject.Call("handleSignIn",credentialResponseResult);
+
+            /// Fetches neccesary data for logging into EOS
+            name = loginObject.Call<string>("getResultName");
+            idToken = loginObject.Call<string>("getResultIdToken");
+
+            /// Invoke Connect Login with fetched Google ID
+            callback.Invoke(idToken, name);
+        }
+
+        /// <summary>
+        /// Failing Callback of GetCredentialAsync  
+        /// GetCredentialAsync is called in com.playeveryware.googlelogin.login.SignInWithGoogle)
+        /// </summary>
+        /// <param name="credentialException"></param>
+        public void onError(AndroidJavaObject credentialException)
+        {
+            loginObject.Call("handleFailure", credentialException);
+        }
+    }
+
+    public class SignInWithGoogleManager : MonoBehaviour
+    {
+        AndroidJavaObject loginObject;
+
+        public void GetGoogleIdToken(System.Action<string, string> callback)
+        {
+            using AndroidJavaClass unityPlayer = new("com.unity3d.player.UnityPlayer");
+            using AndroidJavaObject activity = unityPlayer.GetStatic<AndroidJavaObject>("currentActivity");
+            if (activity == null)
+            {
+                Debug.LogError("EOSAndroid: activity context is null!");
+                return;
+            }
+
+            using AndroidJavaClass loginClass = new AndroidJavaClass("com.playeveryware.googlelogin.login");
+            if (loginClass == null)
+            {
+                Debug.LogError("Java Login Class is null!");
+                return;
+            }
+
+            loginObject = loginClass.CallStatic<AndroidJavaObject>("instance");
+
+            /// Create the proxy class and pass instances to be used by the callback
+            EOSCredentialManagerCallback javaCallback = new EOSCredentialManagerCallback();
+            javaCallback.loginObject = loginObject;
+            javaCallback.callback = callback;
+
+            AndroidConfig config = AndroidConfig.Get<AndroidConfig>();
+            /// SignInWithGoogle(String clientID, String nonce, Context context, CredentialManagerCallback callback)
+            loginObject.Call("SignInWithGoogle", config.GoogleLoginClientID, config.GoogleLoginNonce, activity, javaCallback);
```

Comment:
Seems like config.GoogleLoginClientID and config.GoogleLoginNonce are required. We should consider making validator attributes or running some kind of test on this, or otherwise informing the user if these are null or empty.

Reply:
Added a null check on ClientID.
Nonce can be null.